modules/widgets/simple-payments/customizer.js

```diff
@@ -70,6 +73,31 @@
 		widgetForm.find( '.jetpack-simple-payments-delete-product' ).on( 'click', deleteProduct( widgetForm ) );
 	}
 
+	function syncProductLists() {
+		var request = wp.ajax.post( 'customize-jetpack-simple-payments-buttons-get', {
+			'customize-jetpack-simple-payments-nonce': api.settings.nonce[ 'customize-jetpack-simple-payments' ],
+			'customize_changeset_uuid': api.settings.changeset.uuid
+		} );
+
+		request.done( function( data ) {
+			var selectedProduct = 0;
+
+			$( document ).find( 'select.jetpack-simple-payments-products' ).each( function( index, select ) {
+				var $select = $( select );
+				selectedProduct = $select.val();
+
+				$select
+					.find( 'option' )
+					.remove()
+					.end()
```

Comment:
will this end up jarring the dropdown if someone has it open? could we do a replacement only on the items that need inserting or removal? are we instead ready to accept this rare glitch?

Reply:
So, this is a valid concern, but with the upcoming code freeze this is not the right time to do something too clever.

If we edit a SP product and then immediately (before the widget-sync completes) reopen the dropdown, the worse that can happen is that the dropdown will disappear and reappear, which imho, is not like the highest priority bug ever. 😄

Reply:
I agree that a better reconciliation process between the server and the client would be better, but I'd rather add that later as janitorial task.

Reply:
all good. I mainly want to make sure we're aware of the behaviors we're introducing and not be surprised by them later 👍